File: Array&String/String/Python/reformat_the_string.py
class Solution:
    def reformat(self, s: str) -> str:
        ans, digit, alpha = "", "", ""

        for w in s:
            if w.isalpha(): alpha += w
            else: digit += w

        if abs(len(digit) - len(alpha)) > 1: return ""

        if len(alpha) > len(digit): digit, alpha = alpha, digit

        for i in range(len(alpha)):
            ans += digit[i]
            ans += alpha[i]

        if len(alpha) < len(digit): ans += digit[-1]

        return ans

File: Array&String/String/CPP/reformat_the_string.cpp
#include <algorithm>
#include <cmath>
#include <iostream>
#include <string>
using namespace std;

class Solution {
public:
    string reformat(string s) {
        string digit = "", alpha = "", ans = "";

        for (char x : s) isdigit(x) ? digit += x : alpha += x;

        int p = digit.size(), q = alpha.size();
        if (abs(p - q) > 1) return "";

        if (p < q) {
            swap(digit, alpha);
            swap(p, q);
        }

        ans.reserve(s.size());

        for (size_t  i = 0; i < q; ++i) {
            ans.push_back(digit[i]);
            ans.push_back(alpha[i]);
        }

        if (p > q) {
            ans.push_back(digit[q]);
        }

        return ans;
    }
};

int main() {
    Solution solution;
    string s = "a0b1c2";
    cout << solution.reformat(s) << endl;
    return 0;
}